packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1862,14 +1862,25 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     checkDuplicateExports() {}
 
     parseImport(node: N.Node): N.AnyImport {
-      if (this.match(tt.name) && this.lookahead().type === tt.eq) {
-        return this.tsParseImportEqualsDeclaration(node);
-      }
+      node.importKind = "value";
 
-      if (this.eatContextual("type")) {
-        node.importKind = "type";
-      } else {
-        node.importKind = "value";
+      if (this.match(tt.name)) {
+        const ahead = this.lookahead();
+
+        if (ahead.type === tt.eq) {
+          return this.tsParseImportEqualsDeclaration(node);
+        }
+
+        if (
+          this.state.value === "type" &&
```

Comment:
should also check `!this.state.containsEsc`, which is also included in `isContexual`.
```js
import typ\u{65} typescript from "typescript";
```

Note, it seems that the TS [REPL](https://www.typescriptlang.org/play/?ts=Nightly#code/JYWwDg9gTgLgBDAnmAOgVwN4DYCsBfBZAUwGcBjKYMeAMyghDgCIkxSKqYmBuIA) does not agree with me, which should be a bug.

Reply:
Yeah, let's assume that escapes are disallowed for now and report it to the TS team (cc @DanielRosenwasser)